src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNull.Fixer.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.NetCore.Analyzers.Usage
+{
+    public abstract class DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNullFixer<TInvocationExpression> : CodeFixProvider
+        where TInvocationExpression : SyntaxNode
+    {
+        protected const string HasValue = nameof(Nullable<int>.HasValue);
+        protected const string ArgumentNullException = nameof(System.ArgumentNullException);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                var root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+                if (root.FindNode(context.Span, getInnermostNodeForTie: true) is not TInvocationExpression invocation)
+                {
+                    continue;
+                }
+
+                if (diagnostic.Id == DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNull.NonNullableValueRuleId && invocation.Parent is not null)
+                {
+                    var codeAction = CodeAction.Create(
+                        MicrosoftNetCoreAnalyzersResources.DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNullCodeFixTitle,
+                        _ =>
+                        {
+                            var newRoot = root.RemoveNode(invocation.Parent, SyntaxRemoveOptions.KeepNoTrivia);
+                            if (newRoot is null)
+                            {
+                                return Task.FromResult(context.Document);
+                            }
+
+                            return Task.FromResult(context.Document.WithSyntaxRoot(newRoot));
+                        }, diagnostic.Id);
```

Comment:
NIT: we usually pass the title or the fixer type name, though not sure which one is the suggested one